packages/components/avatar/src/Avatar/Avatar.styles.ts

```diff
@@ -48,23 +59,29 @@ export const getAvatarStyles = ({
       alignItems: 'center',
       justifyContent: 'center',
       fontStretch: 'semi-condensed',
-      fontSize: `${getInitialsFontSize(size)}px`,
+      fontSize: toPixels(getInitialsFontSize(size)),
     }),
     image: css({
-      borderRadius,
+      borderRadius: innerBorderRadius,
       display: 'block',
+
+      // loading skeleton
+      '& + svg': {
+        borderRadius: innerBorderRadius,
+        rect: { rx: 0, ry: 0 }, // has a default 4px border radius
+      },
     }),
     root: css({
       borderRadius,
-      height: finalSize,
+      height: size,
+      width: size,
       overflow: 'hidden',
       position: 'relative',
-      width: finalSize,
-      svg: {
-        borderRadius,
-      },
+      padding: getTotalBorderWidth(colorVariant),
+
+      // color variant border
       '&::after': {
-        borderRadius,
+        borderRadius: borderRadius,
```

Comment:
Nit (or left out for differentiation with `innerBorderRadius`?)

```suggestion
        borderRadius,
```